Create new country budget alignment guidance (v2.03)

PetyaKangalova opened the pull request:
Guidance agreed in here: https://discuss.iatistandard.org/t/guidance-on-implementing-aid-and-budget-alignment/831

The guidance will be added to the current IATI reference page under Overview for the time being. With the development of the new IATI website, all the guidance and consideration pages will be reviewed again and moved to the new IATI website.

@bill-anderson  @markbrough  Please let me know if you want to request any changes to the content.

PetyaKangalova committed:
Create new country budget alignment guidance

Guidance agreed in here: https://discuss.iatistandard.org/t/guidance-on-implementing-aid-and-budget-alignment/831

The guidance will be added to the current IATI reference page under Overview for the time being. With the development of the new IATI website, all the guidance and consideration pages will be reviewed again and moved to the new IATI website.

andylolz commented:
I thiiiink in order to add this page to menus etc, you also need to include `overview/country-budget-alignment` in [the list at the bottom of this page](https://github.com/IATI/IATI-Extra-Documentation/blob/501166bd1584bd2df4b0b9a396acb20fc122ac29/en/activity-standard/overview.rst). Otherwise it won’t be linked from anywhere.

Also, very minor thing, but underlining headings all the way avoids build warnings. See IATI/IATI-Guidance#310.

As a side note… I think it would be great to split out the sectors guidance in here to form part of a “Sectors” overview page. I think that could help make this guidance more obviously grounded in the standard, and therefore more practical. But I think that should form part of the next iteration, because I wouldn’t want to block this guidance from being added.

So apart from the minor changes mentioned above, this all looks fine to me. @bill-anderson @markbrough ?

markbrough commented on en/activity-standard/overview/country-budget-alignment.rst:
```diff
@@ -0,0 +1,40 @@
+This definition was approved by `WP-STAT under the written procedure <http://www.oecd.org/officialdocuments/publicdisplaydocumentpdf/?cote=DCD/DAC/STAT(2015)30/REV1&docLanguage=En/>`_ in February 2016.
+
+
+2) Publish detailed CRS purpose codes
```
There was [a comment on that thread](https://discuss.iatistandard.org/t/guidance-on-implementing-aid-and-budget-alignment/831/17) from @stevieflow that seemed like a good idea -- make more explicit that this should use `sector` rather than `country-budget-items`. So maybe change this line to:

> 2) Publish detailed CRS purpose codes in the `sector` field

markbrough reviewed (approved):
One small suggested change if you agree (to make things slightly more explicit), otherwise think this is good to go. Thanks a lot!!

markbrough commented:
@bill-anderson @dalepotter @PetyaKangalova please can you approve if you are happy -- I think this is good to go, would be great to bring this to a conclusion! Many thanks!!

PetyaKangalova committed:
Small amendment to text

Incorporating change to specify that purpose code should be added to sector field

markbrough commented:
Thanks for approving, @amy-silcock! @PetyaKangalova please do you think you could merge this? And then publish to the IATI Standard website? Thank you!!

PetyaKangalova commented:
@markbrough I have now merged. @dalepotter  would need to update the  IATI-Standard-SSOT with the submodule reference to the new Extra-Documentation and would only be able to do so when he is back in the office on Tuesday (29th of May).

markbrough commented:
Amazing, thanks @PetyaKangalova!!

dalepotter commented on en/activity-standard/overview/country-budget-alignment.rst:
```diff
@@ -0,0 +1,40 @@
+Country Budget Alignment
```
@PetyaKangalova Following our conversion, noting additional points re this:

- As this is a brand new page, it needs adding to the TOC, at: https://github.com/IATI/IATI-Extra-Documentation/blob/version-2.03/en/activity-standard/overview.rst
- It appears that there is content in this page which refers to all versions of the Standard, therefore we need to replicate this guidance so that they appear in other versions too (i.e. make additions to other branches of Extra-Documentation).

dalepotter commented on en/activity-standard/overview/country-budget-alignment.rst:
```diff
@@ -0,0 +1,40 @@
+
+Guidance on implementing budget and aid alignment
+----------
+During the 2.03 upgrade process, decision was made to deprecate BudgetIdentifier codelist. As a result, further guidance was required to provide to publishers on implementing aid and budget alignment.
```
As a minimum I would recommend that the narrative of [`BudgetIdentifier`](https://github.com/IATI/IATI-Codelists-NonEmbedded/blob/master/xml/BudgetIdentifier.xml#L7) gets updated to indicate that it is deprecated at version 2.03. Going forwards, the technical process for how a non-embedded codelist is removed should also be clarified.

andylolz replied:
> As a minimum I would recommend that the narrative of [`BudgetIdentifier`](https://github.com/IATI/IATI-Codelists-NonEmbedded/blob/master/xml/BudgetIdentifier.xml#L7) gets updated to indicate that it is deprecated at version 2.03.

I think removing references to it is even-more-minimum than that. There’s a pull request for that here:
https://github.com/IATI/IATI-Codelists/pull/139

andylolz replied:
> Going forwards, the technical process for how a non-embedded codelist is removed should also be clarified.

Does [the Vocabulary codelist](http://iatistandard.org/203/codelists/Vocabulary/) provide a model for how a codelist (of any sort) can be removed? I.e. remove at the next integer, leaving an explanatory note in its place.

At the moment, the codelist in question is still prominently linked to [on the budget-item documentation page](http://iatistandard.org/203/activity-standard/iati-activities/iati-activity/country-budget-items/budget-item/#attributes). Once that’s removed (IATI/IATI-Codelists#139), the deprecated codelist is only present in the sidebar. Ideally, it would be hidden from there, too (IATI/IATI-Standard-SSOT#141) and then it would be even less likely for a user to stumble across it and get confused.

dalepotter reviewed (commented):
Whilst this PR has been merged to Extra-Documentation, I would recommend it that the submodule reference (in IATI-Standard-SSOT) does not get updated until these issues are resolved.

markbrough commented:
Thanks @dalepotter -- if this could be done in the next few days then it would be great if it could be added to the TOC and if this guidance could be copied to other versions of the Standard - that would clearly make this guidance even more useful. I assume it should be a very quick and easy job to do this. I can send a PR if you want.

However, if it is going to take weeks or months, then please can this be deployed as is. I hope you can understand my impatience, as we are so close and it has already taken over a year to get to this point ;)

markbrough commented:
@dalepotter - this has now been done in #484. Would be great if that PR could be merged and this now deployed? Thank you!

andylolz commented:
I’ve also added this to other versions in #485, #486, #487 and #488.

markbrough commented:
Hi @dalepotter @PetyaKangalova -- I think all the required changes have now been made and merged, so please can this guidance now be deployed? Or let me know if there is something else that needs to be done.

dalepotter commented:
> I think all the required changes have now been made and merged, so please can this guidance now be deployed? Or let me know if there is something else that needs to be done.

@markbrough @andylolz 
All the other PRs (#475 #477 #478 #479) for changes to other versions of the Extra-Documentation have now been approved and merged  - thanks @andylolz for replicating. This means there are no inconsistencies between content at different versions of the Standard relating to this - horray! 

The next step is to update submodules so that these changes are pulled into the SSOT and become an 'official' part of the IATI Standard. However, queue the proverbial rabbit warren as there are issues with running this process locally (at least in an automated way). This has been resolved for v2.03 - thanks again @andylolz - but needs replicating for other versions to be run: see 
https://github.com/IATI/IATI-Standard-SSOT/pull/140#issuecomment-396560295  I can do this, though will be in scheduled maintenance time next Monday, or we would happily accept PRs to do this sooner, to  enable speedier deployment of these changes to the documentation site.
